src/lib/libcore.js

```diff
@@ -2296,6 +2296,74 @@ addToLibrary({
 
   $noExitRuntime: "{{{ makeModuleReceiveExpr('noExitRuntime', !EXIT_RUNTIME) }}}",
 
+  // The following addOn<X> functions are for adding runtime callbacks at
+  // various executions points. Each addOn<X> function has a corresponding
+  // compiled time version named addAt<X> that will instead inline during
+  // compilation (see parseTools.mjs).
+  // Note: if there are both runtime and compile time code, the runtime
+  // callbacks will be invoked before the compile time code.
+
+  // See ATPRERUNS in parseTools.mjs for more information.
+  $__ATPRERUN__: [],
+  $__ATPRERUN____deps: ['$callRuntimeCallbacks'],
+  $__ATPRERUN____postset: () => {
+    ATPRERUNS.unshift('callRuntimeCallbacks(__ATPRERUN__);');
```

Comment:
I find this pretty hard to wrap my head around... but I think I get it.  The naming is very confusing.

Reply:
Yeah, the difference of `at` vs `on` is not easy to reason at from a glance. We could probably rename the compile time versions without much breakage. Maybe a follow up?